src/Storages/MeiliSearch/StorageMeiliSearch.cpp

```diff
@@ -121,26 +121,63 @@ SinkToStoragePtr StorageMeiliSearch::write(const ASTPtr & /*query*/, const Stora
     return std::make_shared<SinkMeiliSearch>(config, metadata_snapshot->getSampleBlock(), local_context);
 }
 
-MeiliSearchConfiguration getConfiguration(ASTs engine_args)
+MeiliSearchConfiguration getConfiguration(ASTs engine_args, ContextPtr context)
 {
-    if (engine_args.size() != 3)
+    if (auto named_collection = getExternalDataSourceConfiguration(engine_args, context))
     {
-        throw Exception(
-            "Storage MeiliSearch requires 3 parameters: MeiliSearch('url', index, key).", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        auto [common_configuration, storage_specific_args] = named_collection.value();
+        String url = common_configuration.addresses_expr;
+        String index = common_configuration.table;
+        String key = common_configuration.password;
+        for (const auto & [arg_name, arg_value] : storage_specific_args)
+        {
+            if (arg_name == "url")
+            {
+                if (!url.empty())
+                    throw Exception(
+                        "2 times given 'url' value, old = " + url + " new = " + arg_value->as<ASTLiteral>()->value.safeGet<String>(),
+                        ErrorCodes::BAD_ARGUMENTS);
+                url = arg_value->as<ASTLiteral>()->value.safeGet<String>();
+            }
+            else if (arg_name == "key" || arg_name == "password")
```

Comment:
It seems now it is not possible to get password from config, only throw kay-value, but the whole point it to be able to get it from config.